src/engine/source/api/src/metrics/handlers.cpp

```diff
@@ -0,0 +1,210 @@
+#include "api/metrics/handlers.hpp"
+
+#include <json/json.hpp>
+#include <eMessages/eMessage.h>
+#include <eMessages/metrics.pb.h>
+
+#include <api/adapter.hpp>
+
+namespace api::metrics::handlers
+{
+
+namespace eMetrics = ::com::wazuh::api::engine::metrics;
+namespace eEngine = ::com::wazuh::api::engine;
+
+/* Manager Endpoint */
+
+api::Handler metricsDumpCmd(const std::shared_ptr<metricsManager::IMetricsManagerAPI>& metricsAPI)
+{
+    return [metricsAPI](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eMetrics::Dump_Request;
+        using ResponseType = eMetrics::Dump_Response;
+        auto res = ::api::adapter::fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        // Validate the params request
+        const auto& eRequest = std::get<RequestType>(res);
+        auto result = metricsAPI->dumpCmd();
+
+        if (std::holds_alternative<base::Error>(result))
+        {
+            return ::api::adapter::genericError<ResponseType>(std::get<base::Error>(result).message);
+        }
+
+        const auto protoVal = eMessage::eMessageFromJson<google::protobuf::Value>(std::get<std::string>(result));
+        const auto json_value = std::get<google::protobuf::Value>(protoVal);
+
+        ResponseType eResponse;
+        eResponse.set_status(eEngine::ReturnStatus::OK);
+        eResponse.mutable_value()->CopyFrom(json_value);
+
+        return ::api::adapter::toWazuhResponse(eResponse);
+    };
+}
+
+
+api::Handler metricsGetCmd(const std::shared_ptr<metricsManager::IMetricsManagerAPI>& metricsAPI)
+{
+    return [metricsAPI](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eMetrics::Get_Request;
+        using ResponseType = eMetrics::Get_Response;
+        auto res = ::api::adapter::fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        // Validate the params request
+        const auto& eRequest = std::get<RequestType>(res);
+        auto errorMsg = !eRequest.has_scopename()        ? std::make_optional("Missing /scope name")
+                        : !eRequest.has_instrumentname() ? std::make_optional("Missing /instrument name")
+                                                         : std::nullopt;
+
+        if (errorMsg.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(errorMsg.value());
+        }
+
+        auto result = metricsAPI->getCmd(eRequest.scopename(), eRequest.instrumentname());
```

Comment:
When searching for a scope that does not exist, `metricsManager::MetricsManager::getCmd` calls `metricsManager::MetricsManager::getScope` and it launches an expiration here:

https://github.com/wazuh/wazuh/blob/1af7899a8f8c5fbd4bce9c9565b7ac34bd665565/src/engine/source/metrics/src/metricsManager.cpp#L113-L114

This is captured by the server, causing the following command to have the wrong output:
![image](https://user-images.githubusercontent.com/1891958/230974194-d5ed274d-848f-4b68-991b-5cdb2550e335.png)